### ISSUE TITLE
Firefox: Focus outline obscures text when link wraps multiple lines

### ISSUE THREAD

**Issue by dengeist** [tatianamac/selfdefined#102]
## Problem
The following screenshots show a link reading "Obsessive Compulsive Disorder (OCD)" as seen on the Self-Defined website, with the site's red custom focus drawn around it.

**In Firefox**, the outline is drawn around _each line_ of the wrapped text, making the text hard to read.
<img width="279" alt="Word_Wrap_Focus_FF" src="https://user-images.githubusercontent.com/13525251/77216860-099eb700-6adb-11ea-928c-2ac39ff9b225.png">

**In Safari and Chrome**, the outline is drawn around the whole anchor element. This way, the outline does not obscure any text. 
<img width="290" alt="Word_Wrap_Focus_Safari" src="https://user-images.githubusercontent.com/13525251/77216931-9f3a4680-6adb-11ea-9ad3-39ca95fd12d3.png">

## Possible fixes 
@wuz (Conlin) suggested using text-decoration styling to restore word-level underline to links when they are rendered as `inline-block`s. Here's [Conlin's codepen demo](https://codepen.io/wuz/pen/BaNPbNb).

The [support matrix for text-decoration styling on caniuse](https://caniuse.com/#feat=text-decoration) looks good.

**Edit:** 
I was initially stoked about Conlin's suggestion, but further research into custom underline properties made me sad. :(
We could also:
- display links as inline-block
- add a child span to them
- apply a bottom border to the child span instead of the link

This approach works in more browsers – even modern browsers do not support custom underlines with text-decoration properties – and is most like our current styles. I made [a CodePen to compare the two approaches](https://codepen.io/dengeist/pen/mdJjYmB).

## Details for nerds
As evidenced in [the Firefox source code](https://searchfox.org/mozilla-central/rev/f1f75f0a2d995241fbc454edfab4be1064c544e1/layout/painting/nsDisplayList.cpp#5389 ), this seems to be a known issue related to how Firefox chooses to draw the focus outline.

**Comment by tatianamac** [comment]
Thank you both for researching this.

What if we add a background color to denote focus instead? 
<img width="698" alt="Word in focus carries pink background" src="https://user-images.githubusercontent.com/7484549/77957685-b66df680-7288-11ea-850b-16d061c0b613.png">

**Comment by wuz** [comment]
@tatianamac That works for me!

**Comment by tatianamac** [comment]
Agree on the `text-decoration` call out (I should have been more explicit). 

@wuz Would you like to open a PR for this?

**Comment by wuz** [comment]
Sure! I can do that tonight!
